core/trino-main/src/test/java/io/trino/operator/output/TestPositionsAppender.java

```diff
@@ -241,14 +252,42 @@ public void testSliceRle()
         }
     }
 
+    @Test
+    public void testSliceAppenderInputNotByteArray()
+    {
+        testAppend(VARCHAR, ImmutableList.of(
+                input(new VariableWidthBlock(3, Slices.wrappedLongArray(257, 2), new int[] {0, 1, Long.BYTES, 2 * Long.BYTES}, Optional.empty()), 0, 2)));
+    }
+
+    @Test
+    public void testSliceAppenderInputRleNotByteArray()
+    {
+        PositionsAppender positionsAppender = new SlicePositionsAppender(1, DEFAULT_MAX_PAGE_SIZE_IN_BYTES);
+        RunLengthEncodedBlock rleBlock = rleBlock(new VariableWidthBlock(1, Slices.wrappedLongArray(77), new int[] {0, Long.BYTES}, Optional.empty()), 10);
+
+        positionsAppender.appendRle(rleBlock);
+
+        long sizeInBytes = positionsAppender.getSizeInBytes();
+        Block actual = positionsAppender.build();
+        assertBlockIsValid(actual, sizeInBytes, VARCHAR, ImmutableList.of(new BlockView(rleBlock, allPositions(rleBlock.getPositionCount()))));
+    }
+
+    @Test
```

Comment:
isn't this already covered by another test? Add comment

Reply:
I don't think so, these tests usually use randomly generated values and an empty string is a specific value that I want to test.
comment added.